Softeq.XToolkit.Common.Tests/Files/BaseFileProviderTests/BaseFileProviderTests.cs

```diff
@@ -0,0 +1,319 @@
+        }
+
+        [Theory]
+        [InlineData(FilePath2)]
+        [InlineData(FilePath3)]
+        public async void OpenFileForWriteAsync_ExistingPath_ReturnsValidStream(string path)
+        {
+            Assert.True(_mockFileSystem.File.Exists(path));
+
+            var stream = await _filesProvider.OpenFileForWriteAsync(path);
+
+            Assert.NotNull(stream);
+        }
+
+        #endregion
+
+        #region ClearFolder
+
+        [Theory]
+        [InlineData(MissingFilePath2)]
+        [InlineData(MissingFilePath3)]
+        [InlineData("")]
+        [InlineData(null)]
+        public async void ClearFolderAsync_MissingOrInvalidPath_ThrowsException(string path)
+        {
+            await Assert.ThrowsAnyAsync<Exception>(async () => await _filesProvider.ClearFolderAsync(path));
+        }
+
+        [Theory]
+        [InlineData(DirectoryPath1)]
+        public async void ClearFolderAsync_ExistingFolderPath_ClearsFolder(string path)
+        {
+            Assert.True(_mockFileSystem.Directory.Exists(path));
+            Assert.NotEmpty(_mockFileSystem.Directory.GetFiles(path));
+
+            await _filesProvider.ClearFolderAsync(path);
+
+            Assert.Empty(_mockFileSystem.Directory.GetFiles(path));
+        }
+
+        #endregion
+
+        #region WriteFile
+
+        [Theory]
+        [InlineData(FilePath2, "content")]
+        [InlineData(FilePath3, "content")]
+        public async void WriteFileAsync_ExistingPath_WritesToFile(
+            string path, string content)
+        {
+            Assert.True(_mockFileSystem.File.Exists(path));
+
+            byte[] byteArray = Encoding.ASCII.GetBytes(content);
+            MemoryStream stream = new MemoryStream(byteArray);
+
+            await _filesProvider.WriteFileAsync(path, stream);
+
+            var fileContent = _mockFileSystem.File.ReadAllText(path);
+            Assert.StartsWith(content, fileContent);
+        }
+
+        [Theory]
+        [InlineData(MissingFilePath2, "content")]
+        [InlineData(MissingFilePath3, "content")]
+        public async void WriteFileAsync_MissingPath_CreatesFileAndWritesToFile(
+            string path, string content)
+        {
+            Assert.False(_mockFileSystem.File.Exists(path));
+
+            byte[] byteArray = Encoding.ASCII.GetBytes(content);
+            MemoryStream stream = new MemoryStream(byteArray);
+
+            await _filesProvider.WriteFileAsync(path, stream);
+
+            Assert.True(_mockFileSystem.File.Exists(path));
+            var fileContent = _mockFileSystem.File.ReadAllText(path);
+            Assert.Equal(content, fileContent);
+        }
+
+        [Theory]
+        [InlineData(null, "content")]
+        [InlineData("", "content")]
+        public async void WriteFileAsync_InvalidPath_ThrowsException(
+            string path, string content)
+        {
+            byte[] byteArray = Encoding.ASCII.GetBytes(content);
+            MemoryStream stream = new MemoryStream(byteArray);
```

Comment:
can be extracted to separate method: `CreateStream(string)`

Reply:
Done